apps/flame-defi/app/features/evm-wallet/services/token-approval-service/token-approval-service.ts

```diff
@@ -0,0 +1,83 @@
+import type { Config } from "@wagmi/core";
+import { type Address, erc20Abi } from "viem";
+import { GenericContractService } from "../generic-contract-service";
+import { EvmCurrency } from "@repo/flame-types";
+
+export class TokenApprovalService extends GenericContractService {
+  constructor(wagmiConfig: Config, contractAddress: Address) {
+    super(wagmiConfig, contractAddress, erc20Abi);
+  }
+
+  async approveToken(
```

Comment:
I think we should return a single allowance here and do the mapping to new allowances where the service is being used so that the approval service isn't worried about state management.